src/hunspell/hashmgr.cxx

```diff
@@ -904,8 +896,23 @@ std::string HashMgr::encode_flag(unsigned short f) const {
     stream << f;
     ch = stream.str();
   } else if (flag_mode == FLAG_UNI) {
-    const w_char* w_c = (const w_char*)&f;
-    std::vector<w_char> w(w_c, w_c + 1);
+
+#if defined(__i386__) || defined(_M_IX86) || defined(_M_X64)
+
+#if __cplusplus >= 202002L
+    auto wc = std::bit_cast<w_char>(f);
+#else
+    w_char wc;
+    memcpy(&wc, &f, sizeof(unsigned short));
+#endif
+
+#else
+    w_char wc;
+    wc.h = (unsigned char)(f >> 8);
+    wc.l = (unsigned char)(f & 0xff);
+#endif
+    //const w_char* w_c = (const w_char*)&f; <- this will not work properly on big endian architecture 
```

Comment:
I'd suggest to just ditch the line rather than keep the commented out code. But I think this encode_flag should be the mirror of decode_flag which is always a memcpy

Reply:
Yes, you are right that `encode_flag` should be the mirror of `decode_flag`, but `decode_flag` should not be `memcpy` on big endian architecture.